core/src/parser/utils.rs

```diff
@@ -619,21 +627,46 @@ pub fn mk_merge_label(src_id: FileId, l: usize, r: usize) -> MergeLabel {
     }
 }
 
-/// Generate a `Let` or a `LetPattern` (depending on whether `assgn` has a record pattern) from
+/// Generate a `Let` or a `LetPattern` (depending on whether there's a binding with a record pattern) from
 /// the parsing of a let definition. This function fails if the definition has both a pattern
-/// and is recursive because recursive let-patterns are currently not supported.
+/// and is recursive because recursive let-patterns are currently not supported. It also fails
+/// if it has multiple bindings and any of them is a pattern.
 pub fn mk_let(
     rec: bool,
-    pat: Pattern,
-    t1: RichTerm,
-    t2: RichTerm,
+    bindings: Vec<LetBinding>,
+    body: RichTerm,
     span: RawSpan,
 ) -> Result<RichTerm, ParseError> {
-    match pat.data {
-        PatternData::Any(id) if rec => Ok(mk_term::let_rec_in(id, t1, t2)),
-        PatternData::Any(id) => Ok(mk_term::let_in(id, t1, t2)),
-        _ if rec => Err(ParseError::RecursiveLetPattern(span)),
-        _ => Ok(mk_term::let_pat(pat, t1, t2)),
+    let all_simple = bindings
+        .iter()
+        .all(|b| matches!(b.pat.data, PatternData::Any(_)));
+
+    if all_simple {
+        Ok(mk_term::let_in(
+            rec,
+            bindings.into_iter().map(|mut b| {
+                let PatternData::Any(id) = b.pat.data else {
+                    unreachable!()
```

Comment:
A justification (even obvious) for why this is unreachable could be nice for future readers.